File: SDKs/amazonaws.com/codecommit/2015-04-13/java/lib/src/main/java/openapisdk/models/shared/PutFileEntry.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * PutFileEntry
 * Information about a file added or updated as part of a commit.
**/
public class PutFileEntry {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("fileContent")
    public String fileContent;
    public PutFileEntry withFileContent(String fileContent) {
        this.fileContent = fileContent;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("fileMode")
    public FileModeTypeEnumEnum fileMode;
    public PutFileEntry withFileMode(FileModeTypeEnumEnum fileMode) {
        this.fileMode = fileMode;
        return this;
    }
    @JsonProperty("filePath")
    public String filePath;
    public PutFileEntry withFilePath(String filePath) {
        this.filePath = filePath;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("sourceFile")
    public SourceFileSpecifier sourceFile;
    public PutFileEntry withSourceFile(SourceFileSpecifier sourceFile) {
        this.sourceFile = sourceFile;
        return this;
    }
}

File: SDKs/amazonaws.com/codecommit/2015-04-13/python/src/sdk/models/shared/putfileentry.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import filemodetypeenum_enum as shared_filemodetypeenum_enum
from ..shared import sourcefilespecifier as shared_sourcefilespecifier


@dataclass_json
@dataclasses.dataclass
class PutFileEntry:
    r"""PutFileEntry
    Information about a file added or updated as part of a commit.
    """
    
    file_path: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('filePath') }})
    file_content: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('fileContent') }})
    file_mode: Optional[shared_filemodetypeenum_enum.FileModeTypeEnumEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('fileMode') }})
    source_file: Optional[shared_sourcefilespecifier.SourceFileSpecifier] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sourceFile') }})
    
